src/core/common/time.hpp

```diff
@@ -0,0 +1,348 @@
+     *
+     */
+    bool operator<(const Time &aOther) const { return ((mValue - aOther.mValue) & (1UL << 31)) != 0; }
+
+    /**
+     * This method indicates whether this `Time` instance is after or equal to another one.
+     *
+     * @param[in]   aOther   A `Time` instance to compare with.
+     *
+     * @retval TRUE    This `Time` instance is after or equal to @p aOther.
+     * @retval FALSE   This `Time` instance is not after or equal to @p aOther.
+     *
+     */
+    bool operator>=(const Time &aOther) const { return !(*this < aOther); }
+
+    /**
+     * This method indicates whether this `Time` instance is before or equal to another one.
+     *
+     * @param[in]   aOther   A `Time` instance to compare with.
+     *
+     * @retval TRUE    This `Time` instance is before or equal to @p aOther.
+     * @retval FALSE   This `Time` instance is not before or equal to @p aOther.
+     *
+     */
+    bool operator<=(const Time &aOther) const { return (aOther >= *this); }
+
+    /**
+     * This method indicates whether this `Time` instance is strictly after another one.
+     *
+     * @param[in]   aOther   A `Time` instance to compare with.
+     *
+     * @retval TRUE    This `Time` instance is strictly after @p aOther.
+     * @retval FALSE   This `Time` instance is not strictly after @p aOther.
+     *
+     */
+    bool operator>(const Time &aOther) const { return (aOther < *this); }
+
+    /**
+     * This static method converts a given number of seconds to milliseconds.
+     *
+     * @returns The number of milliseconds.
+     *
+     */
+    static uint32_t SecToMsec(uint32_t aSeconds) { return aSeconds * 1000u; }
+
+    /**
+     * This static method converts a given number of milliseconds to seconds.
+     *
+     * @returns The number of seconds.
+     *
+     */
+    static uint32_t MsecToSec(uint32_t aMilliseconds) { return aMilliseconds / 1000u; }
+
+private:
+    uint32_t mValue;
+};
+
+/**
+ * This class represents a time instance (millisecond time).
+ *
+ */
+class TimeMilli : public Time
+{
+public:
+    /**
+     * This is the default constructor for a `TimeMilli` object.
+     *
+     */
+    TimeMilli(void)
+        : Time()
+    {
+    }
+
+    /**
+     * This constructor initializes a `TimeMilli` object with a given value.
+     *
+     * @param[in] aValue   The numeric time value to initialize the `Time` object.
+     *
+     */
+    explicit TimeMilli(uint32_t aValue)
+        : Time(aValue)
+    {
+    }
+
+    /**
+     * This constructor initializes a `Time` object from another (base) object.
+     *
+     * @param[in] aTime   A `Time` object.
+     *
+     */
+    TimeMilli(const Time &aTime)
+        : Time(aTime)
+    {
+    }
+
+    /**
+     * This static method returns the current time in milliseconds.
+     *
+     * @returns The current time in milliseconds.
+     *
+     */
+    static TimeMilli GetNow(void) { return Time(otPlatAlarmMilliGetNow()); }
```

Comment:
What was there reason of moving `GetNow()` from `TimerMilli` to here? I think `TimerMilli` is more appropriate since that is the component that has dependency on `otPlatAlarmMilli*`.